src/TLuaInterpreter.cpp

```diff
@@ -17659,3 +17662,118 @@ int TLuaInterpreter::showNotification(lua_State* L)
     mudlet::self()->mTrayIcon.showMessage(title, text, mudlet::self()->mTrayIcon.icon(), notificationExpirationTime);
     return 0;
 }
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#registerMapInfoProvider
+int TLuaInterpreter::registerMapInfoProvider(lua_State* L)
```

Comment:
The following code will make a crash (sorry!):

```lua
registerMapInfoProvider("Red one!", function(roomId, selectionSize, areaId, displayingCurrent) 
  return string.format("RoomID: %d", roomId), false, false, unpack(color_table.red)
end)
```

Reply:
I've just run it and it's OK. Can you give me more details please?

Reply:
It happens to me on the insertion call in line 17684. To be honest that doesn't quite make sense - don't you want to retrieve all values from Lua first, then insert your function?

Reply:
What do you mean by retrieve all values from Lua first?
I am extracting name and callback before insert.
Everything extracted later are returns of callback function called.

Reply:
Right, that makes sense.

Reply:
Does it crash every time? I guess this is unrelated to unpack, maybe something else is going on here? roomId maybe being nil, can you add some debug of arguments coming in?

Reply:
Yep every time, and it is related to unpack somehow - this works fine:

```lua
registerMapInfo("Red one!", function(roomId, selectionSize, areaId, displayingCurrent) 
  return string.format("RoomID: %d", roomId), false, false, 255,255,0
end)
```

This crashes:

```lua
registerMapInfo("Red one!", function(roomId, selectionSize, areaId, displayingCurrent) 
  return string.format("RoomID: %d", roomId), false, false, unpack(color_table.red)
end)
```

Does the unpack work for you?

Reply:
Yes, it does. Will check server build when I am back.